lib/fetch_blocks_list.js

```diff
@@ -0,0 +1,239 @@
+const csv = require('csv-parser');
+const { logger } = require('./logger');
+const { GetObjectCommand, S3Client, ListObjectsCommand } = require('@aws-sdk/client-s3');
+const {
+  S3_KEY,
+  S3_BUCKET,
+  AWS_REGION,
+  AWS_ACCESS_KEY_ID,
+  AWS_SECRET_ACCESS_KEY,
+  EXPORT_BLOCKS_LIST_MAX_INTERVAL } = require('./constants');
+
+
+class BlocksListExporter {
+  constructor() {
+    this.s3Client;
+    this.s3Params;
+    this.blocksList = [];
+  }
+
+  /**
+   * @returns {void} Initializes the S3 client and its parameters
+   */
+  initS3Client() {
+    this.s3Client = new S3Client({
+      credentials: {
+        accessKeyId: AWS_ACCESS_KEY_ID,
+        secretAccessKey: AWS_SECRET_ACCESS_KEY
+      },
+      region: AWS_REGION
+    });
+    this.s3Params = {
+      Bucket: S3_BUCKET
+    };
+  }
+
+  /**
+   * @param {string} key Marker for the next page of results
+   * @returns {Array} Array of the response objects, the last key, and whether the response is truncated
+   */
+  async fetchS3KeysPage(key) {
+    this.s3Params.Marker = key;
+    const data = await this.s3Client
+      .send(new ListObjectsCommand(this.s3Params))
+      .catch((err) => {
+        logger.error(`Error fetching S3 data keys page: ${err}`);
+        throw err;
+      });
+
+    const dataKeys = Object.keys(data);
+    let contents, isTruncated;
+    if (dataKeys.indexOf('Contents') >= 0 && dataKeys.indexOf('IsTruncated') >= 0) {
+      contents = data.Contents;
+      isTruncated = data.IsTruncated;
+    } else {
+      logger.error(`Error fetching S3 data: Missing key in response: ${JSON.stringify(data)}`);
```

Comment:
<img width="934" alt="image" src="https://github.com/santiment/san-chain-exporter/assets/739463/f213b5cd-7f7e-4276-90e5-0b47a96ce6f5">

```
function a() { 
  console.error("Error message");
  throw Error("Error message");             
}
a();
```

This style of logging leads to error message duplicates. In general we should never log before we throw exception. The reason is simple. Most (all) code that catch our exceptions will log stacktrace and error msg upon the catch. Even if we don't catch it ourself, vm will do it for us and print the stacktrace and error message.
I made a small example to show the issue

Reply:
You're right, for this PR I'll remove the redundant logs only from the new code, this is an idea for another PR to remove all such redundant loggings